common/php/trait-block-editor-compatible.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Experiment: a trait that abstracts the logic to overwrite anything that a specific module does.
+ * A prime use case would be not enqueueing EF classic editor assets.
+ *
+ */
+trait Block_Editor_Compatible {
+  protected $ef_module;
+  protected $hooks;
+
+  /**
+   * This method handles init of Module Compat
+   *
+   * @param EF_Module $module_instance
+   * @param array $hooks associative array of hooks and their respective callbacks
+   *
+   * @todo handle priorities too.
+   *
+   * @return void
+   */
+  function __construct( $module_instance, $hooks = [] ) {
+    $this->ef_module = $module_instance;
+    $this->hooks = $hooks;
+    // Highly debatable
```

Comment:
Please expand on the comment :)

Reply:
After debating with myself I removed the comment. Basically, I'm not sure what's the earliest possible hook we can hook into to allow compat modifications to be made.